.gitlab/docker/Dockerfile.spack_compiler

```diff
@@ -0,0 +1,32 @@
+ARG BASE_IMAGE
+FROM $BASE_IMAGE
+
+ARG ARCH
+ARG BOOST
+ARG BUILD_TYPE
+ARG COMPILER
+ARG CXXSTD
+ARG HWLOC
+
+ENV DEBIAN_FRONTEND=noninteractive
+
+SHELL ["/bin/bash", "-c"]
+
+# Disable host compatibility to be able to compile for broadwell on zen2
+RUN echo -e "concretizer:\n  targets:\n    host_compatible: false" \
+    > $SPACK_ROOT/etc/spack/concretizer.yaml
+
+# Install compiler
+RUN export spack_compiler=${COMPILER/clang/llvm} && \
+    spack install $spack_compiler languages=c++ arch=$ARCH && \
+    spack compiler add --scope site $(spack location -i $spack_compiler)
+
+ENV spack_spec "pika@main build_type=$BUILD_TYPE arch=$ARCH %$COMPILER malloc=system cxxstd=$CXXSTD ^$BOOST ^$HWLOC"
```

Comment:
Could this be a single `ARG SPACK_SPEC` instead? Then we'd reduce the number of args, and make it easier to explicitly specify e.g. the fmt version, malloc, etc. in the pipeline config file directly instead of adding an ARG for each dependency.

Reply:
I mean you can still do
```
ARG SPACK_SPEC
ENV SPACK_SPEC=$SPACK_SPEC
```
if you want that behaviour. My comment was mainly about how and where we construct the spec.

Reply:
Or as a dotenv. Whichever you find nicer to work with.